libs/logic-apps-shared/src/parsers/lib/common/schemaprocessor.ts

```diff
@@ -258,7 +258,9 @@ export class SchemaProcessor {
     titlePrefix = this._concatenateString(titlePrefix, schema.title);
     summaryPrefix = this._concatenateString(summaryPrefix, summary);
 
-    if (keys.length && !this._isInternalParameter(schema) && !isReadOnlyInputParameter) {
+    const skipInternal = this.options.excludeInternal && this._isInternalParameter(schema);
+
+    if (keys.length && !skipInternal && !isReadOnlyInputParameter) {
```

Comment:
This isnt right. Now for the internal parameters (legacy) defined in swaggers, any parameter marked as internal should always have a default value.
This function is expanding object properties, if the object has properties it cannot be marked as internal because we should not be parsing it.

While serializing we would be just using the default value to put in the definition for any property marked as internal regardless of its swagger type.